### DISCUSSION TITLE
How to reproduce OOM failures when using FuzzedDataProvider

### DISCUSSION THREAD

**Discussion by centic9** [CodeIntelligenceTesting/jazzer#600] [Q&A] [answered]
Hi, 

When an OOM is produced as part of fuzzing, no reproducer-code is generated. 

When you are using a `byte[]` as fuzz-data, creating a reproducer is easy, but I did not see how to do this when using a `FuzzedDataProvider`.

Is there an example of how to read the data so that it can be used to re-run the fuzzing-target with this data?

When I try to use the Base64 encoded string that is printed together with a `CannedFuzzedDataProvider`, it fails with a `StreamCorruptedException`:

```
        String base64 = "...";

        CannedFuzzedDataProvider provider = new CannedFuzzedDataProvider(base64 /*new String(encode, StandardCharsets.UTF_8)*/);

        Fuzz.fuzzerTestOneInput(provider);

```

**Comment by fmeum** [comment, marked as answer]
We recommend using the [JUnit 5 support](https://github.com/CodeIntelligenceTesting/jazzer#junit-5) to write your fuzz test. Even with OOM crashes, you can run and debug all collected `crash-*` files directly as regular parameterized unit test cases. Existing fuzz tests can be reused with essentially no changes (just add `@FuzzTest` and move them to a test class).

If that doesn't work, you can use the [Jazzer replayer](https://central.sonatype.dev/artifact/com.code-intelligence/jazzer-replayer/0.10.0/versions) to run a given fuzz test class on a given `crash-*` file. We have deprecated it in favor of the JUnit approach, but it should continue to work.

**Reply by centic9** [reply]
Thanks, the @FuzzTest is exactly what I was looking for!
